src/modules/messages.ts

```diff
@@ -0,0 +1,544 @@
+import { DongleConfig } from './DongleDriver'
+
+export enum KeyMapping {
+  invalid = 0, //'invalid',
+  siri = 5, //'Siri Button',
+  mic = 7, //'Car Microphone',
+  left = 100, //'Button Left',
+  right = 101, //'Button Right',
+  frame = 12,
+  selectDown = 104, //'Button Select Down',
+  selectUp = 105, //'Button Select Up',
+  back = 106, //'Button Back',
+  down = 114, //'Button Down',
+  home = 200, //'Button Home',
+  play = 201, //'Button Play',
+  pause = 202, //'Button Pause',
+  next = 204, //'Button Next Track',
+  prev = 205, //'Button Prev Track',
+  wifiEn = 1000,
+  wifiPair = 1012,
+  wifiConnect = 1002,
+}
+
+export type Key = keyof typeof KeyMapping
+
+export enum MessageType {
+  SendFile = 0x99,
+  Open = 0x01,
+  HeartBeat = 0xaa,
+  ManufacturerInfo = 0x14,
+  CarPlay = 0x08,
+  SoftwareVersion = 0xcc,
+  BluetoothAddress = 0x0a,
+  BluetoothPIN = 0x0c,
+  Plugged = 0x02,
+  Unplugged = 0x04,
+  VideoData = 0x06,
+  AudioData = 0x07,
+  Touch = 0x05,
+  BluetoothDeviceName = 0x0d,
+  WifiDeviceName = 0x0e,
+  BluetoothPairedList = 0x12,
+  MediaData = 0x2a,
+}
+
+export class HeaderBuildError extends Error {}
+
+export class MessageHeader {
```

Comment:
@rhysmorgan134 I moved the logic of parsing/constructing messages here - should be easier to write tests for individual messages that way